lib/msprime.c

```diff
@@ -6461,6 +6467,8 @@ msp_set_simulation_model_sweep_genic_selection(msp_t *self, double reference_siz
     /* dt value is expressed in generations for consistency; translate to
      * model time. */
     trajectory->dt = self->model.generations_to_model_time(&self->model, dt);
+
+    ret = msp_rescale_model_times(self);
```

Comment:
Note: this was a bug here. Any times we were getting before will be wrong, unless we happened to always be using Ne=1/4 or something.

Reply:
wait - i've been trying to track down time scaling issues in this implementation for about two weeks now... what will the msp_rescale_model_times to in this case? go from 4N gens to gens? also would this have affected mutation numbers (due to time scaling)? 

been meaning to bother you for help in tracking down these scaling issues but grant reviewing responsibilities have me by the throat at the moment

Reply:
This call scales all times and rates that are in the simulator from generations into model time (4N generations, here, I guess).